texada-src/tests/constinstantspool_test.cpp

```diff
@@ -68,15 +67,15 @@ TEST(ConstInstantsPoolTest,UsableByLinearChecker) {
     ASSERT_EQ(l_valid_instants.at(0).first.at("a"), "a");
     ASSERT_EQ(l_valid_instants.at(0).first.at("c"), "c");
 
-    f->destroy();
+    //f->destroy();
```

Comment:
why is f->destroy removed in this test?

Reply:
SPOT 2.X does not use the destroy function externally like this anymore, a lot of the destruction of formula classes is done internally. In this test, the instantiated_prop_type used in 'valid_instants_on_traces' instantiated the formula by recusing on the formula. In this case, when creating the visitor this is just returning the pointer to the formula so when destroy is called in the function it is also destroying the original formula.